lib/app/migrate.ts

```diff
@@ -0,0 +1,584 @@
+      app =>
+        `${chalk.bold(app.appName)}: ${getUnmigratableReason(app.unmigratableReason)}`
+    );
+
+    throw new Error(
+      `${i18n(`commands.project.subcommands.migrateApp.errors.noAppsEligible`, {
+        accountId: derivedAccountId,
+      })} \n  - ${reasons.join('\n  - ')}`
+    );
+  }
+
+  if (
+    appId &&
+    !allApps.some(app => {
+      return app.appId === appId;
+    })
+  ) {
+    throw new Error(
+      i18n('commands.project.subcommands.migrateApp.prompt.chooseApp', {
+        appId,
+      })
+    );
+  }
+
+  const appChoices = allApps.map(app => ({
+    name: app.isMigratable
+      ? app.appName
+      : `[${chalk.yellow('DISABLED')}] ${app.appName} `,
+    value: app,
+    disabled: app.isMigratable
+      ? false
+      : getUnmigratableReason(app.unmigratableReason),
+  }));
+
+  let appIdToMigrate = appId;
+  if (!appIdToMigrate) {
+    const { appId: selectedAppId } = await listPrompt<MigrationApp>(
+      i18n('commands.project.subcommands.migrateApp.prompt.chooseApp'),
+      {
+        choices: appChoices,
+      }
+    );
+    appIdToMigrate = selectedAppId;
+  }
+
+  const selectedApp = allApps.find(app => app.appId === appIdToMigrate);
+
+  const migratableComponents: string[] = [];
+  const unmigratableComponents: string[] = [];
+
+  selectedApp?.migrationComponents.forEach(component => {
+    if (component.isSupported) {
+      migratableComponents.push(mapToUserFacingType(component.componentType));
+    } else {
+      unmigratableComponents.push(mapToUserFacingType(component.componentType));
+    }
+  });
+
+  if (migratableComponents.length !== 0) {
+    logger.log(
+      i18n('commands.project.subcommands.migrateApp.componentsToBeMigrated', {
+        components: `\n - ${migratableComponents.join('\n - ')}`,
+      })
+    );
+  }
+
+  if (unmigratableComponents.length !== 0) {
+    logger.log(
+      i18n(
+        'commands.project.subcommands.migrateApp.componentsThatWillNotBeMigrated',
+        {
+          components: `\n - ${unmigratableComponents.join('\n - ')}`,
+        }
+      )
+    );
+  }
+
+  logger.log();
+  const proceed = await confirmPrompt(
+    i18n('commands.project.subcommands.migrateApp.prompt.proceed')
+  );
+
+  if (!proceed) {
+    return {};
+  }
+
+  const projectName =
+    name ||
+    (await inputPrompt(
+      i18n('commands.project.subcommands.migrateApp.prompt.inputName')
+    ));
+
+  const { projectExists } = await ensureProjectExists(
+    derivedAccountId,
+    projectName,
+    { forceCreate: false, allowCreate: false, noLogs: true }
+  );
+
+  if (projectExists) {
+    throw new Error(
+      i18n(
+        'commands.project.subcommands.migrateApp.errors.projectAlreadyExists',
+        {
+          projectName,
+        }
+      )
+    );
+  }
+
+  const projectDest =
+    dest ||
+    (await inputPrompt(
+      i18n('commands.project.subcommands.migrateApp.prompt.inputDest'),
+      {
+        defaultAnswer: path.resolve(getCwd(), projectName),
```

Comment:
Make sure to sanitize the project name for the destination path, in case someone decides to be smart and name their project 💩 💩 💩